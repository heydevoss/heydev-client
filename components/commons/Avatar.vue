<template>
  <img :src="avatarUrl || require('~/assets/avatar/default.png')" />
</template>

<script>
export default {
  name: 'Avatar',
  props: {
    avatarUrl: String
  }
}
</script>
<style lang="scss" scoped>
img {
  border-radius: 100%;
  height: 300px;
  object-fit: cover;
  width: 300px;
}
</style>
