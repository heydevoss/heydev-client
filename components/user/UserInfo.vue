<template>
  <div class="content">
    <avatar :avatar-url="avatarUrl" class="user--avatar" />
    <div class="user--info">
      <span class="user--name">{{ name }}</span>
      <span>{{ bio }}</span>
      <span class="user--location">
        <i class="el-icon-location" /> {{ location }}
      </span>
    </div>
  </div>
</template>

<script>
import Avatar from '@/components/commons/Avatar'

export default {
  name: 'UserInfo',
  components: {
    Avatar
  },
  props: {
    name: { type: String, default: '' },
    bio: { type: String, default: '' },
    location: { type: String, default: '' },
    avatarUrl: { type: String, default: '' }
  }
}
</script>

<style lang="scss" scoped>
.content {
  display: flex;
  align-items: center;
}
.user--info {
  margin-left: 40px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}
.user--name {
  font-size: 2rem;
  margin-bottom: 5%;
}
.user--location {
  margin-top: 40px;
}
.user--avatar {
  width: 150px;
  height: 150px;
}

@media screen and (max-width: 820px) {
  .user--avatar {
    width: 60px;
    height: 60px;
  }
}
</style>
