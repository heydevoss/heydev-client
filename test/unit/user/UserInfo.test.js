import { shallowMount } from '@vue/test-utils'
import UserInfo from '@/components/user/UserInfo'
import Avatar from '@/components/commons/Avatar'

describe('<UserInfo/> Component', () => {
  function factory(props) {
    return shallowMount(UserInfo, {
      ...props
    })
  }
  const user = {
    name: 'UserTest',
    bio: 'Hey thats my test',
    location: 'Campina Grande',
    avatarUrl: '~/assets/avatar/default.png'
  }

  const wrapper = shallowMount(UserInfo, {
    propsData: {
      ...user
    },
    stubs: {
      avatar: Avatar
    }
  })

  it('is a Vue instance and renders correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should contain user data', () => {
    expect(wrapper.html()).toContain(user.name)
    expect(wrapper.html()).toContain(user.bio)
    expect(wrapper.html()).toContain(user.location)
    expect(wrapper.html()).toContain(user.avatarUrl)
  })

  it('should render correctly when no props are passed', () => {
    const wrapperWithoutProps = factory()
    expect(wrapperWithoutProps.isVueInstance()).toBeTruthy()
    expect(wrapper.html()).toContain(
      '<span class="user--location"><i class="el-icon-location"></i>'
    )
  })

  it('should set default avatar when its not passed', () => {
    wrapper.setProps({ avatarUrl: null })
    expect(wrapper.html()).toContain('<img src="" class="user--avatar">')
  })
})
