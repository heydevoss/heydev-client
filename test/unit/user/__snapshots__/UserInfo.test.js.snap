// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo/> Component is a Vue instance and renders correctly 1`] = `
<div
  class="content"
>
  <img
    class="user--avatar"
    src="~/assets/avatar/default.png"
  />
   
  <div
    class="user--info"
  >
    <span
      class="user--name"
    >
      UserTest
    </span>
     
    <span>
      Hey thats my test
    </span>
     
    <span
      class="user--location"
    >
      <i
        class="el-icon-location"
      />
       Campina Grande
    
    </span>
  </div>
</div>
`;
